src/applyMiddleware.ts

```diff
@@ -1,8 +1,11 @@
 import compose from './compose'
-import { Middleware, MiddlewareAPI } from './types/middleware'
-import { AnyAction } from './types/actions'
-import { StoreEnhancer, StoreCreator, Dispatch } from './types/store'
-import { Reducer } from './types/reducers'
+import {
+  Middleware,
+  StoreEnhancer,
+  Dispatch,
+  MiddlewareAPI,
+  StoreEnhancerStoreCreator
+} from '.'
```

Comment:
I'd rather keep these as direct imports so readers know where they are actually found in the source code.